### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by skletsun**
## Overview

Fixed warnings of beta/nigtly `clippy`.

### Definition of Done

- [x] There are no TODOs left in the code
- [x] The [coding guidelines](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has documentation
- [x] This change is not breaking **or** mentioned in the Changelog
- [x] The [continuous integration build](https://www.travis-ci.org/jni-rs/jni-rs) passes

### CODE REVIEW

File: src/wrapper/strings/ffi_str.rs [hunk 1]
```diff
@@ -78,8 +78,11 @@ impl JNIString {
 }
 
 impl JNIStr {
-    /// Construct a reference to a `JNIStr` from a pointer. Equivalent to
-    /// `CStr::from_ptr`.
+    /// Construct a reference to a `JNIStr` from a pointer. Equivalent to `CStr::from_ptr`.
+    ///
+    /// # Safety
+    ///
+    /// Expects a valid pointer to a valid immutable C string.
```
Comment (vitvakatu) [inline src/wrapper/strings/ffi_str.rs]: Strings in C can't be mutable, can they? So I suggest
```suggestion
    /// Expects a valid pointer to a null-terminated C string and does not perform any lifetime checks for the resulting value.
```

File: src/wrapper/objects/global_ref.rs [hunk 2]
```diff
@@ -41,8 +41,11 @@ impl<'a> From<&'a GlobalRef> for JObject<'a> {
 }
 
 impl GlobalRef {
-    /// Creates a new global reference. This assumes that `NewGlobalRef`
-    /// has already been called.
+    /// Creates a new global reference.
+    ///
+    /// # Safety
+    ///
+    /// This function assumes that `NewGlobalRef` has already been called.
```
Comment (vitvakatu) [inline src/wrapper/objects/global_ref.rs]: Where is `NewGlobalRef` function in Rust API?

File: src/wrapper/jnienv.rs [hunk 3]
```diff
@@ -83,7 +83,9 @@ pub struct JNIEnv<'a> {
 impl<'a> JNIEnv<'a> {
     /// Create a JNIEnv from a raw pointer.
     ///
-    /// Only does a null check - otherwise assumes that the pointer is valid.
+    /// # Safety
+    ///
+    /// Assumes a valid pointer is retrieved from the `GetEnv` function. Only does a null check.
```
Comment (vitvakatu) [inline src/wrapper/jnienv.rs]: Where is `GetEnv` function in Rust API?

File: src/wrapper/java_vm/vm.rs [hunk 4]
```diff
@@ -166,6 +166,10 @@ impl JavaVM {
     }
 
     /// Create a JavaVM from a raw pointer.
+    ///
+    /// # Safety
+    ///
+    /// Assumes a valid pointer is retrieved from the `JNI_CreateJavaVM` function. Only does a null check.
```
Comment (vitvakatu) [inline src/wrapper/java_vm/vm.rs]: Where is `JNI_CreateJavaVM` function in Rust API?

File: src/wrapper/objects/global_ref.rs (hunk 2 above)
Comment (skletsun) [reply src/wrapper/objects/global_ref.rs]: Specified it's JNI function. The same for other mentions.
P.S. I like the idea it says in comments where to get expected safe value.

File: src/wrapper/java_vm/vm.rs [hunk 5]
```diff
@@ -166,6 +166,11 @@ impl JavaVM {
     }
 
     /// Create a JavaVM from a raw pointer.
+    ///
+    /// # Safety
+    ///
+    /// Assumes a valid pointer is retrieved from the `JNI_CreateJavaVM` JNI function. Only does a
```
Comment (dmitry-timofeev) [inline src/wrapper/java_vm/vm.rs]: Is 'Assumes' a recommended wording for such sections (vs 'Expects' or 'Requires')?

File: src/wrapper/objects/global_ref.rs [hunk 6]
```diff
@@ -41,8 +41,11 @@ impl<'a> From<&'a GlobalRef> for JObject<'a> {
 }
 
 impl GlobalRef {
-    /// Creates a new global reference. This assumes that `NewGlobalRef`
-    /// has already been called.
+    /// Creates a new global reference.
```
Comment (dmitry-timofeev) [inline src/wrapper/objects/global_ref.rs]: That's a little misleading summary — it takes a global reference as `obj` (rename to `global_ref`/`raw_global_ref`/`inner_global_ref`?), and creates a wrapper for it (`GlobalRef`). It does not create new global references (in JNI terms).

Can it be/does it makes sense to make it pub (crate)? It seems no one shall normally use it from the outside.

File: src/wrapper/java_vm/vm.rs (hunk 5 above)
Comment (skletsun) [reply src/wrapper/java_vm/vm.rs]: I think no. Its main purpose is to describe the safety preconditions for an unsafe function.

File: src/wrapper/objects/global_ref.rs (hunk 6 above)
Comment (skletsun) [reply src/wrapper/objects/global_ref.rs]: Made it pub(crate) -> deleted the integration test for testng incorrect usage of JNI.

**Pull request merged**